apps/meteor/client/views/room/body/RoomTopic.tsx

```diff
@@ -0,0 +1,72 @@
+import type { IRoom, IUser } from '@rocket.chat/core-typings';
+import { isTeamRoom } from '@rocket.chat/core-typings';
+import { Box } from '@rocket.chat/fuselage';
+import { RoomBanner, RoomBannerContent } from '@rocket.chat/ui-client';
+import { useSetting, useUserId, useTranslation } from '@rocket.chat/ui-contexts';
+import React from 'react';
+
+import { RoomRoles } from '../../../../app/models/client';
+import MarkdownText from '../../../components/MarkdownText';
+import { usePresence } from '../../../hooks/usePresence';
+import { useReactiveQuery } from '../../../hooks/useReactiveQuery';
+import { RoomLeader } from '../Header/RoomLeader';
+import { useRoomToolbox } from '../contexts/RoomToolboxContext';
+import { useCanEditRoom } from '../contextualBar/Info/hooks/useCanEditRoom';
+
+type RoomTopicProps = {
+	room: IRoom;
+	user: IUser | null;
+};
+
+export const RoomTopic = ({ room, user }: RoomTopicProps) => {
+	const t = useTranslation();
+	const canEdit = useCanEditRoom(room);
+	const userId = useUserId();
+	const directUserId = room.uids?.filter((uid) => uid !== userId).shift();
+	const directUserData = usePresence(directUserId);
+	const useRealName = useSetting('UI_Use_Real_Name') as boolean;
+	const { openTab } = useRoomToolbox();
+
+	const handleAddTopic = () => {
```

Comment:
shouldn't we move to edit room instead?

Reply:
I couldn't find a way, since the route doesn't change from `{room}-settings` on edit. any ideas?

Reply:
I don't think it worth the effort, let's ignore this recommendation!